src/js/background.js

```diff
@@ -1712,17 +1712,13 @@ var tgs = (function() {
         claimTab(tabId);
       }
 
-      // if url has changed
-      if (changeInfo.url) {
-        gsUtils.log(tabId, 'tab url changed. changeInfo: ', changeInfo);
-        checkForTriggerUrls(tab, changeInfo.url);
-        queueSessionTimer();
+      if (tab.url) {
```

Comment:
unless you can point to specific circumstances where this would be actually needed, i don't see the point of keeping it.

Reply:
Since I don't know the inner workings of TMS, I didn't want to change any code that was already there. The main purpose of my change was to add `chrome.tabs.get()`. The rest of the code is the same (just rearranged).

Reply:
i know, but this kind of sloppy programming is a recipe for disaster, or at least unmaintainable code ...

(as things stand, it seems unlikely that someone else will "do a proper job".)

Reply:
Normally I don't just go removing code, even when it appears to be a typo. So I had to look deeper into this to see where `changeInfo.url` was introduced, or at least how long it's been around. I searched as far back as 2017 and it's still there. The code inside the `if ()` statement has changed since then, so maybe in an old browser version `changeInfo.url` actually had a value. But it's been obsolete for a long time, so I think it's safe to remove.

Since I'm still learning GitHub, can you tell me if I can update this PR, or if I need to cancel it and add a new one?

Reply:
i haven't looked deep enough to judge the argument, but at face value it seems fishy - if some code is obsolete, then you should remove it first, in a self-contained commit that clearly explains what exactly (appears) to have been going on. there is a good chance that more related stuff is obsolete, or that the logic was broken entirely.

you apparently managed to update the PR just fine. note however that it's preferable to have a clean history with atomic commits, so massage it with `git rebase -i` (squash away fixups, etc.) and `git push -f` the branch.

Reply:
If you haven’t looked deep enough, why offer an opinion or code review? Saying something is “sloppy” or “fishy” without offering specific and researched recommendations is just lazy IMHO.

Reply:
with sufficient experience, it doesn't take a lot to recognize a job half done.
my review _is_ specific, in that it says that the argumentation isn't conclusive. that doesn't say anything about the change itself, but it should be reason enough to look at it critically.
of course i could do more myself, but we all have our priorities.

Reply:
After looking more at the code around it, simply removing code inside of `if (changeInfo.url)` is not the way to go.

There are 3 calls inside of that if-statement: log(), checkForTriggerUrls(), and queueSessionTimer(). The 1st and 3rd calls are found in the other `chrome.tabs.onEVENT()` callbacks, so they should be kept here to maintain similarity.

@ossilator Thanks for your eye in finding this typo and starting the code review.

In the future, please keep in mind that typos do happen, especially on massive code projects that are constantly evolving. Don't be so quick to look at a bug and jump to conclusions with disparaging characterizations like: sloppy, a recipe for disaster, unmaintainable, fishy, or a job half done. It's repugnant and bad form.

Reply:
why are you speaking of "typo"? there was no talk about anything that would qualify as that.

would you mind rebasing & cleaning up the branch as i suggested? that would make reviewing it properly a lot easier.

Reply:
The typo was `if (changeInfo.url) {`. Instead, it should have been `tab.url`.

Reply:
I believe I got the rebasing working properly. Please have a look now.

Reply:
> The typo is in the line above your code review [...]

i see. but this isn't in fact related to to my previous review.

> I believe I got the rebasing working properly. 

you did on a technical level, but the outcome isn't quite what i had hoped for.
for background, the standard i work with is https://wiki.qt.io/Commit_Policy. while TMS is obviously not bound by it, i think it's a quite reasonable set of rules (including the external pages it references).

so i think yo should do this:
- `git branch for-later`
- `git reset --hard HEAD~2`
- now fix the typo, and commit it
- `git checkout -f for-later -- background.js
- commit it
- `git branch -D for-later`
- `git push -f ...`

i.e., fix the obvious first (and explain what the implications of the bug were), then proceed with the bigger fix (which contains a minor refactoring).

i still think you need to get rid of the "possibly". you need to be able to cite an authoritative reference (a spec, or a commit in chromium), as otherwise you're just poking in the dark. maybe this is in fact a bug in chromium which should be fixed by them (we can still include a workaround here, but it should be clearly designated as such).